src/core/locator/qgslocatorfilter.cpp

```diff
@@ -44,6 +49,92 @@ bool QgsLocatorFilter::stringMatches( const QString &candidate, const QString &s
   return !search.isEmpty() && candidate.contains( search, Qt::CaseInsensitive );
 }
 
+double QgsLocatorFilter::fuzzyScore( const QString &candidate, const QString &search )
```

Comment:
Could you add some unit tests for this please?

Reply:
Sure!

Reply:
I saw some of the tests for `QgsStringUtils` are in python, some in CPP, which one is recommended?

Reply:
It's mostly personal preference. I tend to go Python wherever possible, cos it's quicker to write the tests (no recompilation needed) + means the test doubles as a c++ test + test of the Python bindings too. But in cases where a class already has tests only in c++, I'd usually just extend those instead of creating a new Python test file for it...

Reply:
done